libs/ledger/include/ledger/chain/main_chain.hpp

```diff
@@ -233,16 +234,15 @@ class MainChain
   mutable RMutex   lock_;         ///< Mutex protecting block_chain_, tips_ & heaviest_
   mutable BlockMap block_chain_;  ///< All recent blocks are kept in memory
   // The whole tree of previous-next relations among cached blocks
-  mutable References                references_;
-  TipsMap                           tips_;          ///< Keep track of the tips
-  HeaviestTip                       heaviest_;      ///< Heaviest block/tip
-  LooseBlockMap                     loose_blocks_;  ///< Waiting (loose) blocks
-  std::unique_ptr<BasicBloomFilter> bloom_filter_;
-  bool const                        enable_bloom_filter_;
-  telemetry::GaugePtr<std::size_t>  bloom_filter_queried_bit_count_;
-  telemetry::CounterPtr             bloom_filter_query_count_;
-  telemetry::CounterPtr             bloom_filter_positive_count_;
-  telemetry::CounterPtr             bloom_filter_false_positive_count_;
+  mutable References references_;
+  TipsMap            tips_;          ///< Keep track of the tips
+  HeaviestTip        heaviest_;      ///< Heaviest block/tip
+  LooseBlockMap      loose_blocks_;  ///< Waiting (loose) blocks
+  mutable std::unordered_map<Block::Index, std::unique_ptr<BasicBloomFilter>> bloom_filters_;
```

Comment:
`std::unique_ptr` is probably excess here (I think it even was with the older Bloom filter).

Reply:
It was there as a quick workaround, because the BF is not moveable. It probably should be - fixed.